streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3195,6 +3195,12 @@ object ZStream extends ZStreamPlatformSpecificConstructors with Serializable {
   def fromIterable[A](as: Iterable[A]): Stream[Nothing, A] =
     StreamEffect.fromIterable(as)
 
+  /**
+   * Creates a stream from an effect producing a value of type `Iterable[A]`
+   */
+  def fromIterableEffect[R, E, A](iterable: ZIO[R, E, Iterable[A]]): ZStream[R, E, A] =
+    fromEffect(iterable).flatMap(Stream.fromIterable)
```

Comment:
A bit more efficient:
```suggestion
    fromEffect(iterable).mapConcat(identity)
```